tests/e2e_test.go

```diff
@@ -349,3 +349,127 @@ func TestRelayTxWithWrongNonce(t *testing.T) {
+	batch = append(batch, req_getTransactionCount)
+	// first sendRawTransaction call: rawTx that triggers the error (creates MM cache entry)
+	req_sendRawTransaction := types.NewJsonRpcRequest(1, "eth_sendRawTransaction", []interface{}{testutils.TestTx_MM2_RawTx})
+	batch = append(batch, req_sendRawTransaction)
+	// call getTxReceipt to trigger query to Tx API
+	req_getTransactionReceipt := types.NewJsonRpcRequest(1, "eth_getTransactionReceipt", []interface{}{testutils.TestTx_MM2_Hash})
+	batch = append(batch, req_getTransactionReceipt)
+
+	res, err := testutils.SendBatchRpcAndParseResponse(batch)
+	require.Nil(t, err, err)
+	assert.Equal(t, len(res), 5)
+}
+
+// Test batch request with different transaction
+func TestBatch_CombinationOfSuccessAndFailure(t *testing.T) {
+	resetTestServers()
+
+	var batch []*types.JsonRpcRequest
+	// eth_call intercept
+	req := types.NewJsonRpcRequest(1, "eth_call", []interface{}{map[string]string{
+		"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+		"to":   "0xf1a54b0759b58661cea17cff19dd37940a9b5f1a",
+	}})
+	batch = append(batch, req)
+	// eth_call passthrough
+	req2 := types.NewJsonRpcRequest(1, "eth_callxvssfa", []interface{}{map[string]string{
+		"from": "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+		"to":   "0xf1a54b0759b58661cea17cff19dd37940a9b5f1b",
+	}})
+	batch = append(batch, req2)
+	req_getTransactionCount := types.NewJsonRpcRequest(1, "eth_getTransactionCount", []interface{}{testutils.TestTx_MM2_From, "latest"})
+	batch = append(batch, req_getTransactionCount)
+	// first sendRawTransaction call: rawTx that triggers the error (creates MM cache entry)
+	req_sendRawTransaction := types.NewJsonRpcRequest(1, "eth_sendRawTransactionxxx", []interface{}{testutils.TestTx_MM2_RawTx})
+	batch = append(batch, req_sendRawTransaction)
+	// call getTxReceipt to trigger query to Tx API
+	req_getTransactionReceipt := types.NewJsonRpcRequest(1, "eth_getTransactionReceipt", []interface{}{testutils.TestTx_MM2_Hash})
+	batch = append(batch, req_getTransactionReceipt)
+
+	res, err := testutils.SendBatchRpcAndParseResponse(batch)
+	require.Nil(t, err, err)
+	assert.Equal(t, len(res), 5)
+}
+
+// Test batch request with multiple eth raw transaction
+func TestBatch_Validate_eth_sendRawTransaction_Error(t *testing.T) {
+	resetTestServers()
+	// key=request-id, value=json-rpc error
+	m := make(map[float64]int)
+	m[1] = server.InvalidParams
+	m[2] = server.InvalidParams
+	m[3] = server.InvalidParams
+	m[4] = server.InvalidRequest
+	var batch []*types.JsonRpcRequest
+
+	r1 := types.NewJsonRpcRequest(1, "eth_sendRawTransaction", []interface{}{})     // no params
+	r2 := types.NewJsonRpcRequest(2, "eth_sendRawTransaction", nil)                 // nil params
+	r3 := types.NewJsonRpcRequest(3, "eth_sendRawTransaction", []interface{}{"x"})  // invalid params
+	r4 := types.NewJsonRpcRequest(4, "eth_sendRawTransaction", []interface{}{"xy"}) // invalid request
+	batch = append(batch, r1)
```

Comment:
These 4 lines could be part of a single append call

Reply:
updated